src/assets/TokensController.ts

```diff
@@ -213,102 +260,238 @@ export class TokensController extends BaseController<
     decimals: number,
     image?: string,
   ): Promise<Token[]> {
+    const currentChainId = this.config.chainId;
     const releaseLock = await this.mutex.acquire();
     try {
       address = toChecksumHexAddress(address);
-      const { tokens, ignoredTokens } = this.state;
-      const isERC721 = await this._detectIsERC721(address);
-      const newEntry: Token = { address, symbol, decimals, image, isERC721 };
-      const previousEntry = tokens.find(
+      const { tokens, ignoredTokens, detectedTokens } = this.state;
+      const newTokens: Token[] = [...tokens];
+      const [isERC721, tokenMetadata] = await Promise.all([
+        this._detectIsERC721(address),
+        this.fetchTokenMetadata(address),
+      ]);
+      if (currentChainId !== this.config.chainId) {
+        throw new Error(
+          'TokensController Error: Switched networks while adding token',
+        );
+      }
+      const newEntry: Token = {
+        address,
+        symbol,
+        decimals,
+        image,
+        isERC721,
+        aggregators: formatAggregatorNames(tokenMetadata?.aggregators || []),
```

Comment:
Curious that `aggregators` is the only piece of `tokenMetadata` that gets used. The image isn't used for example, even though it might not be set by the caller. Should we set the `image` if it hasn't already been set?

Reply:
Updated image to be either the image input or fallback to constructing a proxied image similar to how we handle formatting the token list token images.